frontend/simple/js/state.js

```diff
@@ -69,25 +94,45 @@ const mutations = {
     index === -1 && state.pending.push(contractId)
   }
 }
-
 // https://vuex.vuejs.org/en/getters.html
 // https://vuex.vuejs.org/en/modules.html
 const getters = {
   currentGroup (state) {
     return state[state.currentGroupId]
+  },
+  mailbox (state) {
+    return getMailbox(state)
+  },
+  unread (state) {
+    let mailbox = getMailbox(state)
+    return mailbox && state[ mailbox ].messages.filter((msg) => !msg.read).length
   }
 }
 
 const actions = {
+  // Used to update contracts to the current state that the server is aware of
+  async synchronize ({dispatch}: {dispatch: Function}, contractId: string) {
+    let latest = await backend.latestHash(contractId)
+    let recent = await db.recentHash(contractId)
+    if (latest !== recent) {
+      let events = await backend.eventsSince(contractId, recent || contractId)
+      for (let i = 0; i < events.length; i++) {
+        let event = events[i]
+        event.contractId = contractId
+        await dispatch('handleEvent', event)
+      }
+    }
+  },
   async login (
     {dispatch, commit, state}: {dispatch: Function, commit: Function, state: Object},
     user: string
   ) {
     commit('login', user)
     await dispatch('loadSettings')
     await db.saveCurrentUser(user)
-    for (let key of Object.keys(state.contracts)) {
+    for (let key of Object.keys(store.state.contracts)) {
```

Comment:
its stale after the loadSettings is called and therefore the most current state must be used